### PR TITLE
#33 New metric: dependencyPinning

### PR DESCRIPTION
**Pull request by glenscheng**
- added functions for new metric: dependencyPinning
  - src/api/githubApi.ts: functions to get package.json (dependency info) from repos
  - src/metrics/dependencyPinning.ts: functions for scoring based on dependencies in package.json
  - src/types.ts: added new fields
- added unit tests
  - test/index.test.ts
- fixed reading JEST output for logging to console
  - test/test_output.ts
- added all old and new .js to the line coverage calculation for JEST
  - jest.config.js
- `./run test` takes around 2 minutes now
 
image

### CODE REVIEW

**Comment by glenscheng** [comment]
question: are we currently using src/metricCalc.ts? or is it legacy code and we can delete?

File: src/api/githubApi.ts [hunk 1]
```diff
@@ -8,7 +8,7 @@ import {logToFile} from '../utils/log';
 import * as path from 'path';
 import * as fs from 'fs';
 import pLimit = require('p-limit');
-import { sleep } from '../../repos/socketio_socket.io/packages/socket.io-adapter/test/util';
+// import { sleep } from '../../repos/socketio_socket.io/packages/socket.io-adapter/test/util';
```
Comment (jiaccarino) [inline src/api/githubApi.ts:11]: TODO: delete all commented out imports

File: src/metrics/dependencyPinning.js [hunk 2]
```diff
@@ -44,67 +77,99 @@ var isVersionPinned = function (version) {
         if (/^~\d+\.\d+\.\d+$/.test(normalizedVersion)) {
             return true;
         }
-        // Everything else (including ^1.0.0) is considered not pinned
+        // Check for caret ranges with exact minor version (^1.2.3)
```
Comment (jiaccarino) [inline src/metrics/dependencyPinning.js:80]: NIT: Possible implementation change --> if else w/ a flag for T/F

File: src/metrics/dependencyPinning.ts [hunk 3]
```diff
@@ -149,22 +37,54 @@ export const isVersionPinned = (version: string): boolean => {
             return true;
         }
 
-        // Everything else (including ^1.0.0) is considered not pinned
+        // Check for caret ranges with exact minor version (^1.2.3)
```
Comment (jiaccarino) [inline src/metrics/dependencyPinning.ts:40]: NIT: same logic structure here --> return T/F once?

**Review by jiaccarino — APPROVED** [review]
Only nitpicks, rest LGTM!!

**Pull request merged**